compiler/src/dotty/tools/dotc/macros/macros.scala

```diff
@@ -0,0 +1,95 @@
+    def apply(name: String, params: Object*) = {
+      val method = clazz.getDeclaredMethods().find(_.getName == name).get
+      method.setAccessible(true)
+      method.invoke(module, params: _*)
+    }
+  }
+
+  import untpd._
+
+  private def forObject(qual: String)(implicit ctx: Context) = {
+    val clazz = ctx.toolClassloader.loadClass(qual + "$")
+    val module = clazz.getField(nme.MODULE_INSTANCE_FIELD.toString).get(null)
+    new Proxy(clazz, module)
+  }
+
+  /** Whether an Apply tree is quasiquote ?
+   *  @param symbol the symbol of the function in the Apply tree
+   *  @param tree   the Apply tree
+   */
+  def isQuasiquote(symbol: Symbol, tree: Tree)(implicit ctx: Context): Boolean =
+    forObject("scala.meta.eden.quasiquote.package").apply("isQuasiquote", symbol, tree, ctx).asInstanceOf[Boolean]
+
+  /** Expand a quasiquote tree */
+  def expandQuasiquote(tree: Tree, isTerm: Boolean)(implicit ctx: Context): Tree = {
+    forObject("scala.meta.eden.quasiquote.package").apply("expand", tree, Boolean.box(isTerm), ctx).asInstanceOf[Tree]
+  }
+
+  /** Create a transform for macros definitions */
+  def newTransform(implicit ctx: Context): List[MiniPhaseTransform] = {
+    if (!ctx.macrosEnabled) return Nil
+
+    val transform = ctx.toolClassloader.loadClass("scala.meta.eden.transform.MacrosTransform")
+      .newInstance().asInstanceOf[MiniPhaseTransform]
+
+    List(transform)
+  }
+
+  /** Expand annotation macros */
+  def expandAnnot(mdef: MemberDef)(implicit ctx: Context): Tree =
+    forObject("scala.meta.eden.expand.package").apply("expandAnnot", mdef, ctx).asInstanceOf[Tree]
+
+  /** Does the given definition have annotation macros? */
+  def hasAnnotMacros(mdef: MemberDef)(implicit ctx: Context): Boolean = {
+    mdef.mods.annotations.filter(isAnnotMacros).headOption.nonEmpty
+  }
+
+  /** Is the given annotation an annotation macro? */
+  def isAnnotMacros(ann: untpd.Tree)(implicit ctx: Context): Boolean = {
+    import StdNames._
+
+    val symbol = ctx.typer.typedAheadAnnotation(ann)
+    if (!symbol.exists) return false
+
+    val annMethod = symbol.info.decl(nme.apply)
+    val annImplMethod = symbol.owner.info
+      .decl((symbol.name + "$inline").toTermName)
+      .info
+      .decl(nme.apply)
+
+    val macrosAnnotType = ctx.requiredClassRef("scala.annotation.StaticAnnotation")
+
+    symbol.typeRef <:< macrosAnnotType && annMethod.exists && annImplMethod.exists
+  }
+
+  /** Expand def macros */
+  def expandDef(tree: tpd.Tree)(implicit ctx: Context): Tree = {
+    forObject("scala.meta.eden.expand.package").apply("expandDef", tree, ctx).asInstanceOf[Tree]
+  }
+
+  /** Is the given symbol `scalameta.package.apply`? */
+  def isMetaApply(sym: Symbol)(implicit ctx: Context): Boolean = {
+    val meta = ctx.requiredModule("scala.meta.package")
+    meta.info.member("apply".toTermName).symbol == sym
+  }
+
+  def metaTreeType(implicit ctx: Context): Type = {
+    ctx.requiredClassRef("scala.meta.Tree")
```

Comment:
I noticed that in this pull request some `ctx.requiredClassRef` calls are sprinkled around in random places and some are pulled into dedicated methods. What's the convention with respect to that in the Dotty compiler? In scalac, we typically put stuff like that in a dedicated object called `definitions`.